### PR TITLE
[runtime] Fix get_virtual_stelemref_wrapper STELEMREF_CLASS case to actually do the base class check.

### PR DESCRIPTION
**Pull request by kumpera**
The code was bailing straight to native which is significantly slower. This is the result of roslyn compiling corlib on my machine:

Before:
real	0m5.757s
user	0m22.059s
sys	0m1.736s

After:
real	0m5.816s
user	0m23.623s
sys	0m1.693s

### CODE REVIEW

**Comment by akoeplinger** [comment]
did you switch the before/after numbers or did it really get slower?  😄

File: mono/metadata/marshal.c [hunk 1]
```diff
@@ -9975,17 +9973,6 @@ get_virtual_stelemref_wrapper (int kind)
 		/* vklass = value->vtable->klass */
 		load_value_class (mb, vklass);
 
-		/* fastpath */
-		mono_mb_emit_ldloc (mb, vklass);
-		mono_mb_emit_ldloc (mb, aklass);
-		b_fast = mono_mb_emit_branch (mb, CEE_BEQ);
```
Comment (lewurm) [inline mono/metadata/marshal.c]: is that branch never taken?  looks like something we should keep

Comment (lambdageek) [inline mono/metadata/marshal.c]: It's redundant.  klass->supertypes[klass->idepth-1] == klass, always.

**Comment by kumpera** [comment]
@akoeplinger I intentionally swap them to make sure ppl were paying attention. #not

File: mono/metadata/marshal.c (hunk 1 above)
Comment (lewurm) [inline mono/metadata/marshal.c]: ah! 👍  thanks 🙂

**Comment by lewurm** [comment]
builddeb

**Pull request merged**